zebra-state/src/service/block_iter.rs

```diff
@@ -1,102 +1,86 @@
 //! Iterators for blocks in the non-finalized and finalized state.
```

Comment:
This file contains a complete rewrite, so it will be easier to review side-by-side.